fenix/app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -107,11 +110,21 @@ fun FaviconListItem(
         description = description,
         onClick = onClick,
         beforeListAction = {
-            Favicon(
-                url = url,
-                size = ICON_SIZE,
-                modifier = Modifier.padding(horizontal = 16.dp),
-            )
+            if (faviconPainter != null) {
+                Image(
+                    painter = faviconPainter,
+                    contentDescription = null,
+                    modifier = Modifier
+                        .padding(horizontal = 16.dp)
+                        .size(ICON_SIZE),
+                )
+            } else {
+                Favicon(
+                    url = url,
+                    size = ICON_SIZE,
+                    modifier = Modifier.padding(horizontal = 16.dp),
+                )
+            }
```

Comment:
I'm wondering if it's worth propagating this pattern all the way down into `Favicon` in a follow-up. I don't really know enough about how often they get invalidated to know if it's safe to rely on properties cached in memory.